### PR TITLE
Update pkg_manager.erb to include support/detection for AmazonLinux 2 and AmazonLinux 2022

### PR DESCRIPTION
**Pull request by imphocused**
This provides detection for AmazonLinux 2022 (dnf) and fallback 'yum' support for Amazon Linux 2, as well as other RHEL'ish clones which use /etc/system-release.

### CODE REVIEW

File: app/views/unattended/provisioning_templates/snippet/pkg_manager.erb
Comment (ekohl) [inline app/views/unattended/provisioning_templates/snippet/pkg_manager.erb:26]: Out of interest: which other systems use `/etc/system-release`? And we already source `/etc/os-release` if it's present. Can't we rely on those variables?

**Review by ekohl — COMMENTED** [review]
Could you also update the comment that starts on line 16?

File: app/views/unattended/provisioning_templates/snippet/pkg_manager.erb
Comment (imphocused) [reply app/views/unattended/provisioning_templates/snippet/pkg_manager.erb:26]: /etc/os-release is available, but the snippet doesn't  recognize yum in Amazon Linux 2. 

The current RHEL clones support /etc/system-release (alma,rocky,amzn). Fedora and RHEL obviously but those already have explicit support/detection.

I found the reference from this post on Redhat, [Proper way to extract Product, Release, Update, etc](https://access.redhat.com/discussions/2798501) which discusses OS detection

_3) System Release, vendor string: /etc/system-release (also via symlink, or legacy, /etc/fedora-release, /etc/redhat-release)_

The response links to a [sysconfcollect module](url) that identifies some 50+ distros near the bottom of the page. I didnt check any of them because I don't use them, though I don't expect support for them in Foreman/Katello, as I don't imagine them seeing much use (if any) in Foreman installations.

Comment (stejskalleos) [reply app/views/unattended/provisioning_templates/snippet/pkg_manager.erb:26]: If we detect `/etc/system-release` can we use `dnf` instead of the `yum`?
Tested on Rocky & Alma and they both have `dnf`.

**Review by stejskalleos — CHANGES_REQUESTED** [review]
Can you update the commit message? _Update pkg_manager.erb_ is just too generic and doesn't say much about the actual changes.

File: app/views/unattended/provisioning_templates/snippet/pkg_manager.erb
Comment (ekohl) [reply app/views/unattended/provisioning_templates/snippet/pkg_manager.erb:26]: Amazon Linux 2 still has yum since it's EL7 based. Amazon Linux 2022 is based on Fedora so it has dnf.

**Comment by stejskalleos** [comment]
@ekohl Changes LGTM, do you have any other comments?

**Comment by ekohl** [comment]
It still needs a Redmine issue and the commit should refer to it. Otherwise we can't merge it (our changelog generation relies on it). Other than that I'd be happy to accept it.

**Review by stejskalleos — CHANGES_REQUESTED** [review]
@imphocused I see that you changed the title in github but the commit message is still same: `Update pkg_manager.erb`

Commit message should be in format `Fixes #35670 - Description of changes`, where the number is id of the issue in our tracker system.

I created the [#35670 issue](https://projects.theforeman.org/issues/35670) for you, so please just update the message, for example like this: `Fixes #35670 - pkg_manager.erb - support AmazonLinux 2 & 2022`

**Comment by imphocused** [comment]
I've updated and pushed the change, sorry for the delay.

**Comment by stejskalleos** [comment]
ok to test

**Comment by stejskalleos** [comment]
LGTM from my side, @ekohl any other comments?

**Review by ekohl — CHANGES_REQUESTED** [review]
I really hate it when comments are out of sync with the real code so I'd still appreciate it if the comment was updated (or removed):
https://github.com/theforeman/foreman/blob/b323ceca04e37b3958ffa3fc60c098f78098cbfd/app/views/unattended/provisioning_templates/snippet/pkg_manager.erb#L16-L24

**Comment by stejskalleos** [comment]
@imphocused any updates? Updating comment is last thing before we can merge

**Review by stejskalleos — APPROVED** [review]
:green_apple: **LGTM**

**Comment by stejskalleos** [comment]
@ekohl any other comments? There are two commits now but I'll squash them before the merge

**Review by ekohl — APPROVED** [review]

**Pull request merged**
